cdap-app-templates/cdap-dq/src/main/java/co/cask/cdap/dq/DataQualityApp.java

```diff
@@ -137,6 +96,26 @@ public ConfigClass(int workflowScheduleMinutes, DataQualitySource source,
   @Override
   public void configure() {
     ConfigClass configObj = getContext().getConfig();
+    Preconditions.checkArgument(configObj.workflowScheduleMinutes > 0,
+                                "Workflow Frequency in minutes (>0) should be provided");
+    Preconditions.checkArgument(configObj.source != null, "Configuration for DataQualityApp Source is missing");
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(configObj.source.getName()),
+                                "Data Quality source name should not be null or empty");
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(configObj.source.getId()),
+                                "Data Quality source id should not be null or empty");
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(configObj.datasetName),
+                                "Output Dataset name should be not be null or empty");
+    Preconditions.checkArgument(configObj.fieldAggregations != null, "Field Aggregations needs to be specified");
```

Comment:
'fieldAggregations' instead of Field Aggregations, since that is the name of the config field